### PR TITLE
Feature 11582 tested pathogen on pathogen test

### LINKED ISSUE #11582: Add tested pathogen attribute to pathogen test [3]
### Feature Description

This issue is part of the EPIC https://github.com/hzi-braunschweig/SORMAS-Project/issues/11533 on the environmental surveillance module. Environment samples are tested for multiple pathogens and not for specific diseases; therefore, pathogen tests need to be extended to have a "Tested pathogen" attribute.

### Acceptance Criteria

- [x] A "Tested pathogen" field has been added to pathogen tests in both the web and mobile app, containing all pathogens in the customizable enum table
- [x] The new field is shown in pathogen test forms for pathogen tests linked to environment samples; it is hidden for normal samples; vice versa, the "Tested disease" field is hidden for pathogen tests linked to environment samples
- [x] The field is mandatory for environment samples
- [x] The following columns introduced in #11566 have been implemented:
  - Positive pathogen tests: A list of all tested pathogens of all pathogen tests related to the environment sample with a positive laboratory result
  - Latest pathogen test: The tested pathogen and result of the latest pathogen test linked to the environment sample
- [x] Add a new line to the environment samples list in the mobile app containing the pathogens that the sample has been positively tested for
- [x] The tested pathogen filter in the environment sample grid has been properly implemented (i.e. it shows all environment sample with at least one pathogen test with the specified tested pathogen)

### Additional Information

PATHOGEN TEST FORM

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #11582

### CODE REVIEW

**Comment by sergiupacurariu** [comment]
the testedPathogen field is missing from Pathogentest table

**Review by sergiupacurariu — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java [hunk 1]
```diff
@@ -101,6 +119,19 @@ public EnvironmentSampleGrid(EnvironmentSampleCriteria criteria) {
...
 		((Column<EnvironmentSampleIndexDto, String>) getColumn(EnvironmentSampleIndexDto.LABORATORY)).setMaximumWidth(200);
+		((Column<EnvironmentSampleIndexDto, List<Pathogen>>) getColumn(EnvironmentSampleIndexDto.POSITIVE_PATHOGEN_TESTS))
+			.setRenderer(new TextRenderer() {
+
+				@Override
+				public JsonValue encode(Object value) {
+					if (value != null) {
+
+						return super.encode(
+							String.join(", ", ((List<Pathogen>) value).stream().map(t -> t.getCaption()).collect(Collectors.toList())));
```
Comment (sergiupacurariu) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java:130]: the value is null conducting to NullPointerException

**Review by sergiupacurariu — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java (hunk 1 above)
Comment (leventegal-she) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java:130]: there is the null check above

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12576

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/samples/environmentsample/EnvironmentSampleGrid.java:130]: this row still generates a NullPointerException despite the fact that the "value" is not null but it is empty.

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12576

**Review by sergiupacurariu — APPROVED** [review]

**Pull request merged**